m4/netcdf.m4

```diff
@@ -27,6 +29,8 @@ AC_DEFUN([CONFIGURE_NETCDF],
         ])
 
   AS_CASE("${netcdfversion}",
+          [system], [PKG_CHECK_MODULES([NETCDF], [netcdf >= 4.2])],
```

Comment:
Do we need to make a local copy of pkg.m4 before we try to use PKG_CHECK_MODULES?  My system has it installed, but we don't want bootstrap to rely on anything more than our own contrib/autotools version.

Reply:
It is part of pkgconf, so it should be there already.

From several years of packaging experience, bundled copies only get out of sync with upstream and are a reoccurring PITA.

Reply:
> It is part of pkgconf, so it should be there already.

Sadly I have committed the ought-is fallacy and been burned by it too many times to not check, and ... it's almost guaranteed to be available on Ubuntu (It's a dependency for *ubuntu-desktop* now, of all things?  The default GUI now depends on autoconf macros??), but the next system I checked was an OSX laptop, where although everything I need to build (and even bootstrap) libMesh is there, but pkg.m4 isn't.

> bundled copies only get out of sync with upstream and are a reoccurring PITA.

I can't disagree with this.  But adding dependencies willy-nilly has gotten me in trouble too.  What does pkg.m4 itself depend on?  If it's self-contained (and looks like it is, in a couple hundred lines?) I'd feel safer working with a copy of it than letting people without it break.